src/bert.erl

```diff
@@ -11,9 +11,19 @@
 %% API functions
 %%====================================================================
 
+%%@doc Encode Erlang term to a binary data.
+
+-spec encode(Term) -> Result when
+	Term :: term(),
+	Result :: binary().
```

Comment:
I think the correct way of writing this spec would be:
```erlang
-spec encode(Term :: term()) -> Result :: binary().
```
I think the spec may be left out, since dialyzer should be able to infer types for `term_to_binary` and `encode_term`. Same for `decode`.

Reply:
Yes. It is straightforward when we read a code. While we read documentation on hex.pm (for instance) it is not. So for documentation perposes it could be nice and very usefult to add `-spec` for public API.

Reply:
Various styles of spec are possible. I suggest the most readable for html-documentation.

Reply:
Edoc displays the function documentation differently (I would say better) when using the `when` syntax, but exdoc just dumps all the spec, which makes the `when` version a bit worse.

I've checked rebar3 and cowboy documentations, and they use the `-spec func(type1(), type2()) -> type3()` style. Would the following be a nice tradeoff?
```erlang
-spec encode(term()) -> binary().
```

Reply:
> I've checked rebar3 and cowboy documentations, and they use the `-spec func(type1(), type2()) -> type3()` style. Would the following be a nice tradeoff?
> 
> ```erlang
> -spec encode(term()) -> binary().
> ```
Yes, sure.